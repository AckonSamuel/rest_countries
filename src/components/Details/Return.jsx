import { useNavigate } from 'react-router-dom';
import { useTheme } from '@mui/material/styles';
import Paper from '@mui/material/Paper';
import Button from '@mui/material/Button';
import ArrowBackIcon from '@mui/icons-material/ArrowBack';
import Typography from '@mui/material/Typography';
import pxToRem from '../../assets/theme/pxToRem';

const Return = () => {
  const navigate = useNavigate();

  const theme = useTheme();

  const goHome = () => {
    navigate('/');
  };

  return (
    <Paper width="100%" sx={{ marginBottom: pxToRem(60), marginTop: pxToRem(50) }}>
      <Button
        variant="returnHome"
        component="button"
        onClick={() => goHome()}
        sx={{
          width: pxToRem(136),
          backgroundColor: theme.palette.background.primary,
          height: pxToRem(40),
          justifyContent: 'space-evenly',
        }}
      >
        <ArrowBackIcon />
        <Typography variant="returnHome">Back</Typography>
      </Button>
    </Paper>
  );
};

export default Return;
