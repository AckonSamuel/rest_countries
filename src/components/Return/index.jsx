import { useNavigate } from 'react-router-dom';
import { useTheme } from '@mui/material/styles';
import Paper from '@mui/material/Paper';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import ArrowBackIcon from '@mui/icons-material/ArrowBack';
import Typography from '@mui/material/Typography';
import pxToRem from '../../assets/theme/pxToRem';

const Return = () => {
  const navigate = useNavigate();

  const theme = useTheme();

  const goHome = () => {
    navigate('/');
  };

  return (
    <Box
      width="100%"
      sx={{
        marginBottom: pxToRem(60),
        marginTop: pxToRem(50),
        [theme.breakpoints.down('lg')]: {
          marginBottom: pxToRem(0),
          marginTop: pxToRem(40),
        },
      }}
    >
      <Paper sx={{
        width: pxToRem(136),
        [theme.breakpoints.down('md')]: {
          width: pxToRem(104),
          height: pxToRem(30),
          fontSize: pxToRem(14),
        },
      }}
      >
        <Button
          variant="returnHome"
          component="button"
          onClick={() => goHome()}
          sx={{
            width: pxToRem(136),
            backgroundColor: theme.palette.background.primary,
            height: pxToRem(40),
            justifyContent: 'space-evenly',
            [theme.breakpoints.down('md')]: {
              width: pxToRem(104),
              height: pxToRem(30),
              fontSize: pxToRem(14),
            },
          }}
        >
          <ArrowBackIcon />
          <Typography
            sx={{
              [theme.breakpoints.down('md')]: {
                fontSize: pxToRem(14),
              },
            }}
            variant="returnHome"
          >
            Back
          </Typography>
        </Button>
      </Paper>
    </Box>
  );
};

export default Return;
