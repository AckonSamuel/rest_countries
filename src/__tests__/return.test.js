import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Return from '../components/Details/Return';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Return />
      </BrowserRouter>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
