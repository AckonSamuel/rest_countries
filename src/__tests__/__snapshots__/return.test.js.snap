// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root css-17i0y7w"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-16a289p-MuiPaper-root"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-returnHome MuiButton-returnHomePrimary MuiButton-sizeMedium MuiButton-returnHomeSizeMedium MuiButton-root MuiButton-returnHome MuiButton-returnHomePrimary MuiButton-sizeMedium MuiButton-returnHomeSizeMedium css-1pjyywg-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ArrowBackIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
      <span
        className="MuiTypography-root MuiTypography-returnHome css-s74x2d-MuiTypography-root"
      >
        Back
      </span>
    </button>
  </div>
</div>
`;
